### DISCUSSION TITLE
[A-L] (2023/24) Foglio 7 - Esercizio 3

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/algebra#150] [Esercizi - Viaggi/Piazza]
![Screenshot 2023-11-12 150757](https://github.com/sapienzastudentsnetwork/algebra2324/assets/107944476/8730562f-9f1f-4083-8488-397ee9ad1950)

**Comment by cornflexxx** [comment]
Partiamo dicendo che sicuramente  $Id_{\mathbb{Z} _n} \in Aut(\mathbb{Z} _n)$ , ovvero l'applicazione identica da $\mathbb{Z} _n \to \mathbb{Z} _n$. Come già è stato dimostrato un  omomorfismo mappa sempre l'elemento neutro nell'elemento neutro, quindi possiamo aspettarci che il gruppo degli automorfismi contenga  delle applicazioni che fanno corrispondere i due elementi neutri e poi permutano gli altri elementi.  Questi tipi di applicazioni sono suriettive quando fanno corrispondere un generatore ad un altro generatore del gruppo. 

--------------------
Esempio in $\mathbb{Z} _6$:

Consideriamo un'applicazione  $\phi ([1])\to[5]$, sappiamo che questa applicazione è suriettiva perché i soli multipli  di $[5]$ ci permettono di mappare tutto $\mathbb{Z} _6$, infatti $[z]\in\mathcal{U}(\mathbb{Z} _n)\Rightarrow$ è un generatore del gruppo. Quindi svolgendo i calcoli dell'intero periodo di $[5]$ otteniamo la seguente mappa: 

$$ \begin{bmatrix}
1 &  2&  3&  4&  5&6  \\
 5&4  &3  &2  &1  & 6
\end{bmatrix} $$ 

Più in generale $\phi$  rappresenta un insieme di endomorfismi suriettivi che mappano un generatore in un altro generatore del gruppo. Quindi sono permutazioni  che spostano un elemento $[z]\in\mathbb{Z} _n\to[uz]$, dove $u \in \mathcal{U}(\mathbb{Z} _n)$. In effetti anche l'applicazione identità fa questo passaggio : $Id([z])=[1]\cdot[z]$ con $[1]\in\mathcal{U}(\mathbb{Z} _n)$. Quindi in generale tutti gli omomorfismi del tipo $\phi_{u}\in Aut(\mathbb{Z} _n)$ con $u\in\mathcal{U}(\mathbb{Z} _n)$. Tali omomorfismi oltre ad essere suriettivi sono anche iniettivi, infatti possiamo verificare che l'elemento neutro ovvero $[0]=[n]$ è l'unico ad essere mappato da $\phi_u$ in se stesso (in tutte le permutazioni che facciamo non viene mai "toccato"), quindi il $Ker\phi_u =$ { $[n]$ }.

------------------
Quindi ad $Aut(\mathbb{Z} _n)$ appartengono tutte le funzioni così definite:

$$\phi_u:\begin{array}{rcl}
\mathbb{Z}_n & \longrightarrow & \mathbb{Z}_n  \\
\phi([z]) & \longrightarrow & [uz]
\end{array}\\quad \text{con $u\in\mathcal{U}(\mathbb{Z}_n)$}$$

Ma allora $|Aut(\mathbb{Z} _n)|=|\mathcal{U}(\mathbb{Z}_n)|=\varphi(n)$

------------------------
Verifica di  $(Aut(\mathbb{Z} _n),\circ)$:

- L'operatore $\circ$ è ben definito infatti: $\phi_u\circ\phi_{u'}([z])=\phi_u([u'z])=[uu'z]$, essendo $u,u',z\in\mathbb{Z}_n\Rightarrow [uu'z]\in\mathbb{Z}_n$
- Per $\circ$ vale la proprietà associativa
- $\phi_u \circ Id([z]) =\phi_u([z])=Id\circ\phi_u([z])$
- $\forall \phi_u\in Aut(\mathbb{Z_n})\quad\exists\quad\phi_{u'}\circ\phi_u([z])=Id([z])$, per dimostrarlo ci basta svolgere l'operazione e notare che : $\phi_{u'}([uz])=[u'uz]$, sappiamo che   $u,u'\in\mathcal{U}(\mathbb{Z} _n)$, quindi $u$ ammette un inverso in $\mathbb{Z}_n$.


$(Aut(\mathbb{Z} _n),\circ)$ è abeliano : $\phi_u\circ\phi _{u'}=[uu'g]=[u'ug]=\phi _{u'}\circ\phi_u$ (dovuto al fatto che anche gli interi modulo n rispetto alla somma sono un gruppo commutativo)

**Comment by CiottoloMaggico** [comment]
![image](https://github.com/sapienzastudentsnetwork/algebra2324/assets/79638284/5796dc86-75a9-4928-a248-93a0c8489cb6)
